pkg/autoscaler/stats_reporter.go

```diff
@@ -249,24 +246,19 @@ func (r *Reporter) ReportRequestedPodCount(v int64) error {
 	return r.report(requestedPodCountM.M(v))
 }
 
-// ReportActualPodCount captures value v for actual pod count measure.
-func (r *Reporter) ReportActualPodCount(v int64) error {
-	return r.report(actualPodCountM.M(v))
-}
-
-// ReportNotReadyPodCount captures value v for not ready pod count measure.
-func (r *Reporter) ReportNotReadyPodCount(v int64) error {
-	return r.report(notReadyPodCountM.M(v))
-}
-
-// ReportPendingPodCount captures value v for pending pod count measure.
-func (r *Reporter) ReportPendingPodCount(v int64) error {
-	return r.report(pendingPodCountM.M(v))
-}
-
-// ReportTerminatingPodCount captures value v for terminating pod count measure.
-func (r *Reporter) ReportTerminatingPodCount(v int64) error {
-	return r.report(terminatingPodCountM.M(v))
+// ReportActualPodCount captures values for ready, not ready, terminating, and pending pod count measure.
+func (r *Reporter) ReportActualPodCount(ready, notReady, terminating, pending int64) error {
+	var err error
+	if err = r.report(actualPodCountM.M(ready)); err != nil {
```

Comment:
```suggestion
	if err := r.report(actualPodCountM.M(ready)); err != nil {
```

Apply to all others as well.